### DISCUSSION TITLE
Group sidebar items in autogenerated sidebars?

### DISCUSSION THREAD

**Discussion by IanVS** [PaloAltoNetworks/docusaurus-openapi-docs#418] [Q&A] [answered]
I'm trying to group the sidebar items by tag, but it seems that the `sidebar.js` is ignored when using docusaurus autogenerated sidebars, isn't it?  Should the plugin generate the .mdx files into a hierarchical folder structure based on tag?  Or, is there another way to achieve nested sidebars?

**Comment by sserrata** [comment, marked as answer]
Hi @IanVS, the grouping by tag is currently only supported through the generated `sidebar.js` slice and would be incompatible with `autogenerated` sidebars (which are based on directory structure). It's an "either or" at this point.  Writing an a hierarchical folder structure sounds like an interesting approach but I am not sure how it would be materially different from generating the `sidebar` slice. Can you elaborate?

**Reply by IanVS** [reply]
I'm adding the api docs into an existing documentation site that is using autogenerated sidebar entries. Is it possible to mix in the sidebars fine that this library generates?

**Reply by sserrata** [reply]
Yes, it's definitely ok to mix the generated sidebar slice with `autogenerated` sidebar items, non-API doc items, etc. See the demo `sidebars.js` for reference to how this could work.

https://github.com/PaloAltoNetworks/docusaurus-openapi-docs/blob/main/demo/sidebars.js#L50

**Comment by Maschga** [comment]
Hi,
First of all thank you very much for this hard work!

Are there currently efforts to implement a hierarchical folder structure, @sserrata ?
This would help me a lot.
